twilight-http-ratelimiting/src/in_memory/bucket.rs

```diff
@@ -162,10 +164,10 @@ impl BucketQueue {
     }
 
     /// Receive the first incoming ratelimit request.
-    pub async fn pop(&self, timeout_duration: Duration) -> Option<TicketNotifier> {
+    pub async fn pop(&self) -> Option<TicketNotifier> {
         let mut rx = self.rx.lock().await;
 
-        timeout(timeout_duration, rx.recv()).await.ok().flatten()
+        rx.recv().await
```

Comment:
I am somewhat certain that this will never return `None` (as atleast one sender is tied to each `BucketQueue`), which is otherwise necessary for the bucket to be finished and freed (see `BucketQueueTask::run`).